src/test/regress/expected/global_cancel.out

```diff
@@ -1,6 +1,7 @@
 CREATE SCHEMA global_cancel;
 SET search_path TO global_cancel;
 SET citus.next_shard_id TO 56789000;
+SET citus.max_adaptive_executor_poolsize TO 1;
```

Comment:
should be `citus.max_adaptive_executor_pool_size` ?

Reply:
Also, after we re-connect via `\c - global_cancel_user - :master_port` (or any `\c`, do we need to re-set? Not sure, if the flaky part is there, we could consider.

Reply:
another option could be to use:

```SQL
SET citus.grep_remote_commands TO '%whatever we want to print%';
```

Because I'm really not sure if executor pool size helps here? Aren't we already using a single connection for `pg_cancel_backend `  on the remote node?

Reply:
> Also, after we re-connect via `\c - global_cancel_user - :master_port` (or any `\c`, do we need to re-set? Not sure, if the flaky part is there, we could consider.

The flaky part is before we do any `\c` commands. So I think we do not need to set it again.

I am unable to reproduce this on my local, so I do not have an exact idea on whether it solves our issue or not.

Here is the failing diff where we have an extra rollback issued to a worker node:
```diff
 BEGIN;
 SELECT pg_cancel_backend(:coordinator_gpid) FROM dist_table WHERE a = 1;
 NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, 6145, '2022-03-31 19:51:38.619341-07');
 DETAIL:  on server postgres@localhost:57637 connectionId: 4
 NOTICE:  issuing SELECT pg_cancel_backend('13700080000019081'::bigint) AS pg_cancel_backend FROM global_cancel.dist_table_56789000 dist_table WHERE (a OPERATOR(pg_catalog.=) 1)
 DETAIL:  on server postgres@localhost:57637 connectionId: 4
+NOTICE:  issuing ROLLBACK
+DETAIL:  on server postgres@localhost:57637 connectionId: 4
 ERROR:  canceling statement due to user request
 END;
```

Reply:
> Here is the failing diff where we have an extra rollback issued to a worker node:

So, then it is probably not related to pool size, right? It is more around timing on when rollback is sent?